samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentFacadeQueries.java

```diff
@@ -724,8 +729,33 @@ public List<AssessmentFacade> getBasicInfoOfAllActiveAssessmentsByAgent(String o
 			if (questionSizeMap.get(a.getAssessmentBaseId()) != null) {
 				questionSize = ((Long) questionSizeMap.get(a.getAssessmentBaseId())).intValue();
 			}
-			AssessmentFacade f = new AssessmentFacade(a.getAssessmentBaseId(), a.getTitle(), a.getLastModifiedDate(), a.getStartDate(), a.getDueDate(), a.getReleaseTo(), releaseToGroups, lastModifiedBy, questionSize);
-			assessmentList.add(f);
+
+			if (releaseToGroups != null) {
+				try {
+					Site site = SiteService.getSite(siteAgentId);
+					Collection<Group> siteGroups = site.getGroupsWithMember(userId);
```

Comment:
are you sure all of this is cached? if you have 100 assessments, this seems like a huge amount of extra work

Reply:
you should be doing work like this exactly once, not 100 times inside a loop

Reply:
Solved